### PR TITLE
fix: missing elevated card shadow on dark mode

### LINKED ISSUE #5893: Add a shadow to an elevated Card component in dark mode
### On which framework/platform would you like to see this feature implemented?

React

### Which UI component is this feature-request for?

Primitive components

### Please describe your feature-request in detail.

The Card component, even as previewed in the [documentation](https://ui.docs.amplify.aws/react/components/card), lacks a shadow in dark mode, and is visually identical with the default card. Please make the behavior consistent between light and dark modes.

cards

### Please describe a solution you'd like.

_No response_

### We love contributors! Is this something you'd be interested in working on?

- [ ] 👋 I may be able to implement this feature request.
- [ ] ⚠️ This feature might incur a breaking change.

### PR DESCRIPTION
**Pull request by tiffanynwyeung**
#### Description of changes
This change fixes missing box shadows on the elevated variation of the `Card` component. This is because dark mode shadow colors are not defined so it defaults to the light mode shadows, which are extremely difficult to spot in dark mode.

This change also visually changes/affects the dark mode versions of other components that use box shadows in general: `Authenticator`, `Menu`, and `Slider Field`.

- adds dark box shadow colors to dark mode override

Component | Before | After |
:----: | :----: | :----:
Card |  | 
Authenticator |  | 
Menu |  | 
Slider Field (thumb) |  | 

#### Issue #, if available
Fixes #5893.

#### Checklist

- [x] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [x] PR description included
- [x] `yarn test` passes and tests are updated/added
- [x] PR title and commit messages follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) syntax
- [x] _If this change should result in a version bump_, [changeset added](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) (This can be done after creating the PR.) This does not apply to changes made to `docs`, `e2e`, `examples`, or other private packages.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/ui/src/theme/defaultDarkModeOverride.ts [hunk 1]
```diff
@@ -104,6 +104,12 @@ const darkModeTokens = {
       tertiary: '{colors.neutral.20}',
     },
 
+    shadow: {
+      primary: { value: 'hsla(0, 0%, 0%, 0.75)' },
```
[the remark this answers is not in the document]
Comment (dbanksdesign) [reply packages/ui/src/theme/defaultDarkModeOverride.ts:108]: LGTM!

File: packages/ui/src/theme/defaultDarkModeOverride.ts [hunk 2]
```diff
@@ -104,6 +104,12 @@ const darkModeTokens = {
       tertiary: '{colors.neutral.20}',
     },
 
+    shadow: {
+      primary: { value: 'hsla(100, 100%, 100%, 0.25)' },
+      secondary: { value: 'hsla(100, 100%, 100%, 0.15)' },
+      tertiary: { value: 'hsla(100, 100%, 100%, 0.05)' },
```
Comment (jordanvn) [inline packages/ui/src/theme/defaultDarkModeOverride.ts:110]: Nit: Is this tertiary value being used anywhere besides the slider? The alpha of 0.05 makes sense to match the light mode counterpart, but it's so subtle in dark mode that I can't even see it. Would it make sense to consider increasing the tertiary alpha value for dark mode?

**Review by jordanvn — APPROVED** [review]
Looks good to me. Tertiary shadow is a bit hard for me to see on dark mode, but the slider already has a very subtle shadow to begin with, so could just be my opinion. Not blocking

**Review by dbanksdesign — APPROVED** [review]
LGTM!

**Pull request merged**